src/compiler/checker.ts

```diff
@@ -12161,6 +12164,29 @@ module ts {
             return false;
         }
 
+        function checkGrammarTypeQuery(node: TypeQueryNode): boolean {
+
+            if (node && node.exprName) {
+
+                if (checkGrammarTypeQueryExpression(node.exprName)) {
+                    return grammarErrorOnNode(node, Diagnostics.Identifier_expected);
+                }
+                return false;
+            }
+
+            return true;
+        }
+
+        function checkGrammarTypeQueryExpression(node: Expression): boolean {
+            if (node.kind === SyntaxKind.PropertyAccessExpression) {
+                return checkGrammarTypeQueryExpression((<PropertyAccessExpression>node).name);

```

Comment:
you want to recurse down the 'expression' side of the PropertyAccessExpression, not the 'name' side.